### PR TITLE
Add Vagrant tests and resolve userxattr

### LINKED ISSUE #82: VM tests on Github Actions
Run tests with vagrant VMs on github actions to be able to test nested mounts and the other bugfixes.  

Things that we want to test

SELinux?
Other distros
Basic LVM setups (with /home mounted separately, and other nested mounts) https://github.com/binpash/try/issues/19
More curl | bash scripts

### PR DESCRIPTION
**Pull request by ericzty**
- [X] Address parts of #104 RE: (user)xattr.
- [X] Debian
- [X] Debian w/ LVM
- [X] Ubuntu
- [X] Fedora
- [X] run tests
- [x] Rustup test
- [x] Rebase after #99 and remove exit edit to `try`

Closes #82

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
+Vagrant.configure("2") do |config|
+
+  #Reguar debian testing box
```
Comment (angelhof) [inline Vagrantfile:3]: typo :)

**Review by angelhof — APPROVED** [review]
Looks good to me! Let's merge after @mgree reviews

File: .github/workflows/vagrant.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Vagrant
```
Comment (mgree) [inline .github/workflows/vagrant.yaml:1]: Might be good to have a more descriptive name (VM/distro tests?)

File: .github/workflows/vagrant.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+jobs:
+  debian:
```
Comment (mgree) [inline .github/workflows/vagrant.yaml:10]: This is exactly what build matrices are for.

```
strategy:
  matrix:
    vagrant_target:
      - debian
      - debianrustup
      - debianlvm
      - rocky9
      - fedora33

  jobs:
    vagrant:
      runs-on: macos-latest
      steps:
        - name: Checkout
        - uses: actions/checkout@v2
         
        - name: Run VM in Vagrant
        - run: vagrant up ${{ matrix.vagrant_target }}
```

**Review by mgree — CHANGES_REQUESTED** [review]
One tiny tweak. Looks great!

**Pull request merged**